src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -372,7 +372,24 @@ public long GetSize()
 
         try
         {
-            return _db.Get(key);
+            Span<byte> data = _db.GetSpan(key);
+            if (data.IsNull())
+            {
+                return null;
+            }
+            try
+            {
+                if (data.Length == 0)
```

Comment:
Maybe if length is 0, it still need to free it?

Reply:
Will still free it as happens in the `finally` block